src/scripts/main.js

```diff
@@ -1 +1,66 @@
 'use strict';
+
+document.addEventListener('contextmenu', (ev) => {
+  ev.preventDefault();
+});
+
+const first = new Promise((resolve, reject) => {
+  document.addEventListener('click', (ev) => {
+    if (ev.button === 0) {
+      resolve('First promise was resolved');
+    }
+  });
+
+  setTimeout(() => reject(new Error('First promise was rejected')), 3000);
+});
+
+const second = new Promise((resolve) => {
+  document.addEventListener('mousedown', (ev) => {
+    if (ev.button === 0 || ev.button === 2) {
+      resolve('Second promise was resolved');
+    }
+  });
+});
+
+let leftClick = false;
+let rightClick = false;
+
+const third = new Promise((resolve, reject) => {
+  document.addEventListener('mousedown', (ev) => {
+    if (ev.button === 0) {
+      leftClick = true;
+    }
+
+    if (ev.button === 2) {
+      rightClick = true;
+    }
+
+    if (leftClick && rightClick) {
+      resolve('Third promise was resolved');
+    }
+  });
+});
```

Comment:
The 'third' promise resolves when both left and right mouse buttons are pressed, but it does not reset the `leftClick` and `rightClick` flags after resolving. This means it will not resolve again if both buttons are pressed a second time. Consider resetting these flags after resolving the promise if repeated resolution is desired.